tests/server/test_agents_api.py

```diff
@@ -0,0 +1,246 @@
+    await create_agent(name="paused_agent", paused=True)  # paused
+    await create_agent(name="paused_agent_with_instance", paused=True, with_process=True)  # paused
+    await create_agent(name="unpause_on_resume", unpause_on_resume=True)  # down
+    await create_agent(
+        name="failover1", with_process=True, last_failover=(datetime.datetime.now() - datetime.timedelta(minutes=1))
+    )  # up
+    await create_agent(name="failover2", with_process=True, last_failover=datetime.datetime.now())  # up
+
+
+@pytest.mark.asyncio
+async def test_agent_list_filters(client, environment, env_with_agents):
+    result = await client.get_agents(environment)
+    assert result.code == 200
+    assert len(result.result["data"]) == 9
+
+    # Test status filters
+    result = await client.get_agents(environment, filter={"status": "down"})
+    assert result.code == 200
+    assert len(result.result["data"]) == 2
+    assert all([agent["status"] == "down" for agent in result.result["data"]])
+
+    result = await client.get_agents(environment, filter={"status": "up"})
+    assert result.code == 200
+    assert len(result.result["data"]) == 5
+    assert all([agent["status"] == "up" for agent in result.result["data"]])
+
+    result = await client.get_agents(environment, filter={"status": "paused"})
+    assert result.code == 200
+    assert len(result.result["data"]) == 2
+    assert all([agent["status"] == "paused" for agent in result.result["data"]])
+
+    result = await client.get_agents(environment, filter={"status": ["paused", "up"]})
+    assert result.code == 200
+    assert len(result.result["data"]) == 7
+    assert all([agent["status"] == "paused" or agent["status"] == "up" for agent in result.result["data"]])
+
+    result = await client.get_agents(environment, filter={"name": "with_instance"})
+    assert result.code == 200
+    assert len(result.result["data"]) == 4
+    assert all(["with_instance" in agent["name"] for agent in result.result["data"]])
+
+    result = await client.get_agents(environment, filter={"process_name": "failover"})
+    assert result.code == 200
+    assert len(result.result["data"]) == 2
+    assert all(["failover" in agent["name"] for agent in result.result["data"]])
+
+    result = await client.get_agents(environment, filter={"name": "agent", "status": "down"})
+    assert result.code == 200
+    assert len(result.result["data"]) == 1
+
+
+def agent_names(agents):
+    return [agent["name"] for agent in agents]
+
+
+@pytest.mark.parametrize(
+    "order_by_column, order",
+    [
+        ("name", "DESC"),
+        ("name", "ASC"),
+        ("status", "DESC"),
+        ("status", "ASC"),
+        ("process_name", "DESC"),
+        ("process_name", "ASC"),
+        ("last_failover", "DESC"),
+        ("last_failover", "ASC"),
+        ("paused", "DESC"),
+        ("paused", "ASC"),
+        ("unpause_on_resume", "DESC"),
+        ("unpause_on_resume", "ASC"),
+    ],
+)
```

Comment:
You can stack the `pytest.mark.parametrize` annotations to obtain all the combination:
```suggestion
@pytest.mark.parametrize("order_by_column", ["name", "status", "process_name", "last_failover", "paused", "unpause_on_resume"])
@pytest.mark.parametrize("order", ["DESC", "ASC"])
```

Reply:
That's very nice, I didn't know that. Thank you :+1: